crates/cheatnet/src/cheatcodes/spy_events.rs

```diff
@@ -18,33 +20,50 @@ pub enum SpyOn {
     Multiple(Vec<ContractAddress>),
 }
 
+impl SpyOn {
+    pub fn should_be_spied(&mut self, contract_address: ContractAddress) -> bool {
+        match self {
+            SpyOn::All => true,
+            SpyOn::One(address) => *address == contract_address,
+            SpyOn::Multiple(addresses) => addresses.contains(&contract_address),
+        }
+    }
+}
+
 impl CheatnetState {
-    pub fn spy_events(&mut self, spy_on: SpyOn) {
-        self.cheatcode_state.spy_events = Some(spy_on);
+    pub fn spy_events(&mut self, spy_on: SpyOn) -> usize {
+        self.cheatcode_state.spies.push(spy_on);
+        self.cheatcode_state.spies.len() - 1
```

Comment:
Why `-1`?

Reply:
Because we add it to the vector and want to get it's index then